src/dev/impl/DevToys/ViewModels/Tools/EncodersDecoders/JwtDecoderEncoder/JwtDecoder.cs

```diff
@@ -78,17 +78,26 @@ public class JwtDecoder
             TokenParameters tokenParameters,
             TokenResult tokenResult)
         {
-            SigningCredentials? signingCredentials = GetValidationCredentials(tokenParameters);
             var validationParameters = new TokenValidationParameters
             {
-                ValidateIssuerSigningKey = true,
-                IssuerSigningKey = signingCredentials.Key,
-                TryAllIssuerSigningKeys = true,
                 ValidateActor = decodeParameters.ValidateActor,
                 ValidateLifetime = decodeParameters.ValidateLifetime,
                 ValidateIssuer = decodeParameters.ValidateIssuer,
                 ValidateAudience = decodeParameters.ValidateAudience
             };
+            
+            if (decodeParameters.ValidateIssuerSigningKey)
+            {
+                SigningCredentials? signingCredentials = GetValidationCredentials(tokenParameters);
+                validationParameters.ValidateIssuerSigningKey = decodeParameters.ValidateIssuerSigningKey;
+                validationParameters.IssuerSigningKey = signingCredentials.Key;
```

Comment:
If `signingCredentials` is null, this will crash.

Reply:
I believe this is ok for two reasons.
1. The same could be said for old code too, so I have not introduced a new problem.
2. It won't "crash"; instead, the exception will get caught and propagated to the UI. In this case, it will tell the user that the key is null, so they must enter a key if they want to validate the token. Again, that is the same as the existing behavior.

![image](https://github.com/veler/DevToys/assets/7417301/140efcfa-f28b-4c40-8989-25de8520a04e)

(Technically the error is from `GetValidationCredentials`, which should either throw or return a non-nullable. So I guess we could remove the nullable here to alleviate your concern. But again I was mostly refactoring the existing code.)